control/prometheus.py

```diff
@@ -0,0 +1,308 @@
+#
+#  Copyright (c) 2021 International Business Machines
+#  All rights reserved.
+#
+#  SPDX-License-Identifier: LGPL-3.0-or-later
+#
+#  Authors: pcuzner@ibm.com
+#
+
+import os
+import time
+import logging
+import threading
+import spdk.rpc as rpc
+
+from prometheus_client.core import REGISTRY, GaugeMetricFamily, CounterMetricFamily, InfoMetricFamily
+from prometheus_client import start_http_server, GC_COLLECTOR
+from typing import NamedTuple
+
+COLLECTION_ELAPSED_WARNING = 0.8   # Percentage of the refresh interval before a warning message is issued
+REGISTRY.unregister(GC_COLLECTOR)  # Turn of the garbage collector metrics
+
+
+class RBD(NamedTuple):
+    pool: str
+    namespace: str
+    image: str
+
+
+def timer(method):
+    def call(self, *args, **kwargs):
```

Comment:
Not a big deal, but with decorators one [should do this](https://docs.python.org/3/library/functools.html#functools.wraps), so that the returned method has all the metadata/types/signature/docstring/... of the decorated `method`:

```suggestion
    @functools.wraps(method)
    def call(self, *args, **kwargs):
```